src/wasm/kotlin/src/arcs/Particle.kt

```diff
@@ -10,55 +10,143 @@ abstract class Particle : WasmObject() {
     private val toSync: MutableSet<Handle> = mutableSetOf()
     private val eventHandlers: MutableMap<String, (Map<String, String>) -> Unit> = mutableMapOf()
 
+    /** Execute on initialization of Particle. */
+    open fun init() = Unit
+
+    /**
+     * Associate a handle name to a handle object.
+     *
+     * @param name Name of handle from particle in manifest
+     * @param handle Singleton or Collection, defined in this particle class
+     */
     fun registerHandle(name: String, handle: Handle) {
         handle.name = name
         handle.particle = this
         handles[name] = handle
         log("Registering $name")
     }
 
-    fun eventHandler(name: String, handler: (Map<String, String>) -> Unit) {
-        eventHandlers[name] = handler
-    }
-
-    fun connectHandle(handleName: String, willSync: Boolean): Handle? {
-        log("Connect called internal '$handleName'")
+    /**
+     * Connect to a registered handle
+     *
+     * If a handle has been previously registered, return the handle. Optionally, mark the handle for later
+     * synchronization.
+     *
+     * @param name Name of the handle
+     * @param willSync Mark handle for synchronization; depends if handle is readable
+     * @return The name-associated handle, or null
+     * @see [registerHandle]
+     * @see [onHandleSync]
+     */
+    fun connectHandle(name: String, willSync: Boolean): Handle? {
```

Comment:
Can we make this the same API as the wasm/C++ please?

`connectHandle(name: String,  canRead: Boolean, canWrite: Boolean)`

Reply:
Implemented; Direction support will be impl later